docs/backend_platform_support_matrix.md

```diff
@@ -0,0 +1,96 @@
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+# Backend-Platform Support Matrix
+
+Even though Triton supports inference across varied platforms such as
+cloud, data center, edge and embedded devices on NVIDIA GPUs, x86 and
+ARM CPU, or AWS Inferentia, it does so by relying on the backends.
+Note that not all Triton backends support all these different
+platforms. The purpose of this document is to go over what all
+compute platforms are supported by each of these Triton backends.
+GPU in this document refers to Nvidia GPU. See
+[GPU, Driver, and CUDA Support Matrix](https://docs.nvidia.com/deeplearning/frameworks/support-matrix/index.html)
+to learn more about supported GPUs.
+
+## Ubuntu 20.04
+
+The table below describes target device supported for inference by
+each backend on multiple systems.
+
+| Backend      | x86-GPUs  | ARM-SBSA-GPUs | x86 CPU only | ARM-SBSA CPU only |
+| ------------ | --------- | ------------- | ------------ |------------------ |
+| TensorRT     |    GPU    |     GPU       |  Unsupported |  Unsupported      |
+| ONNX Runtime |  CPU/GPU  |   CPU/GPU     |     CPU      |      CPU          |
+| TensorFlow   |  CPU/GPU  |   CPU/GPU     |     CPU      |      CPU          | 
+| PyTorch      |  CPU/GPU  |   CPU/GPU     |     CPU      |      CPU          |
+| OpenVINO     |    CPU    |     CPU       |     CPU      |      CPU          |
+| Python       |CPU/GPU[^1]|  CPU/GPU[^1]  |     CPU[^1]  |      CPU[^1]      |
+| DALI         |   GPU     |     GPU       |  Unsupported |  Unsupported      |
+| FIL          |  CPU/GPU  |  Unsupported  |    CPU       |  Unsupported      |
+
+
```

Comment:
I think it would be good to add the definition of CPU-only too (i.e. a container that doesn't have the CUDA libraries).

Reply:
With Ryan's suggestion below, this should not be needed I guess. It will simplify our support.